cli/src/cli.mjs

```diff
@@ -470,18 +487,52 @@ if ( parsed.help ) {
     });    
   }
 
-}
-  else if (parsed.getStatus){
-    axios.get(url(host, port, '/api/v1/status'), undefined)
-      .then(function (response) {
-        console.log(response.data);
-      })
-      .catch(function (error) {
-        logError(error);
+} else if (parsed.getStatus) {
+  axios.get(url(host, port, '/api/v1/status'), undefined)
+    .then(function (response) {
+      console.log(response.data);
+    })
+    .catch(function (error) {
+      logError(error);
     });
-  }
-  else {
-
+} else if (parsed.downloadOverrides) {
+  /* overrideDirectory is mock-firebolt/cli/externalOverrides by default, 
+  if overrided via CLI, will have user given location to save repository contents */
+  const overrideDirectory = parsed.overrideLocation ? parsed.overrideLocation : getAppRootDir() + '/cli/externalOverrides'
+  // git repository url to be downloaded
+  const downloadUrl = parsed.downloadOverrides;
+
+  // cloning git repository into a temporary folder using tmp node library
+  msg(`Downloading github repository ${downloadUrl}...`);
+  const tmpobj = tmp.dirSync();
+  const tmpFolder = (tmpobj.name).replace(/\\/g, "/");
+  shell('git -C ' + tmpFolder + ' clone ' + downloadUrl + '').then((res) => {
+
+    // copying the file contents inside temporary folder to overrideDirectory, once successfully cloned
+    if (res.code == 0) {
+      var files = fs.readdirSync(tmpobj.name)[0];
+      shell('rm -rf ' + tmpFolder + '/' + files + '/.git').then(() => {
+        shell('cp -R ' + tmpFolder + '/' + files + '/. ' + overrideDirectory).then((res) => {
+          msg(`Copying downloaded contents to ${overrideDirectory}`);
+
+          // removing the temporary folder, once successfully copied
+          if (res.code == 0) {
+            shell('rm -rf ' + tmpFolder).then((res) => {
```

Comment:
Let's do this either way. Whether we copied successfully or not we can clean up the temporary files.